pex/common.py

```diff
@@ -503,11 +523,12 @@ def copy(self, src, dst, label=None):
         but with a different label.
         """
         dst = self._normalize(dst)
-        self._tag(dst, label)
-        self._ensure_parent(dst)
-        shutil.copy(src, os.path.join(self.chroot, dst))
+        if self._tag(dst, label):
+            self._ensure_parent(dst)
+            shutil.copy(src, os.path.join(self.chroot, dst))
```

Comment:
This is a potential behavior change. Before a second instance of the file would overwrite, now 1st wins. I think in reality dup handling is not exercised and in fact most of the ceremony of Chroot can be nuked (please do not do this! - keep the PR focused). Is all this new self._tag conditional checking just to save time on ensure parent? I'm sure that is very very far in the noise, but you could hyperfine some benchmarks on small PEXes which is surely the only place it has a chance of showing up. If it doesn't show up - simpler is better here given the surmise is correct that dups don't happen in practice and the perf is in the noise.

Reply:
Ah! I hadn't realized this!